src/elements/x-send-transaction/x-send-transaction.js

```diff
@@ -336,19 +334,30 @@ export default class XSendTransaction extends MixinRedux(XElement) {
             return;
         }
 
-        // TODO Skip network request when doing airgapped tx creation
         if (address) {
+            // Already accept the recipient as valid and check the recipient type in background
+            this._validRecipient = true;
+            this._clearError('recipient');
+            if (Config.offline) return;
             if (!this.properties.hasConsensus) {
-                if (Config.offline) {
-                    this._setError('Cannot validate address in offline mode', 'recipient');
-                    this._validRecipient = true;
-                } else {
-                    this._setError('Cannot validate address (not connected). Retrying...', 'recipient');
-                    this.__validateRecipientTimeout = setInterval(this._validateRecipient.bind(this), 1000);
-                    this._validRecipient = true;
-                }
+                this._validateRecipientTimeout = setTimeout(this._validateRecipient.bind(this), 300);
             } else {
-                this.__debouncedValidateRecipient(address);
+                // background recipient type check
+                networkClient.client
+                    .then((client) => client.getAccountTypeString(address))
+                    .then((accountType) => {
+                        const validRecipientType = accountType === 'basic';
+                        this._validRecipient = this._validRecipient && validRecipientType;
+                        if (!validRecipientType) {
+                            this._setError('Can only send to Basic Addresses', 'recipient');
```

Comment:
This error message might not be optimal. Addresses are all the same, even the ones for contracts are also _addresses_. There is no _basic_ or _non-basic_ address. That is the account type. Since we cannot use _account_ here, because that is something else in our interface, I would prefer to say here:
```
This address is a contract, which cannot receive new funds.
```